src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -884,6 +954,27 @@ private static String findModuleName(Path modInfoPath) {
             }
         }
 
+        // returns the endianness of the target platform, if the target platform is known
+        // and supported for creating an image through jlink. Else returns null.
+        private static ByteOrder getNativeEndianOfTargetPlatform(String targetPlatform) {
+            int index = targetPlatform.indexOf("-"); // of the form <operating system>-<arch>
+            if (index < 0) {
+                // unknown arch
+                return null;
+            }
+            String archName = targetPlatform.substring(index + 1);
+            return switch (archName) {
+                case "x86", "x86_64",
+                        "amd64", "arm", "aarch64",
+                        "loongarch64", "ppc64le",
+                        "riscv32", "riscv64" -> ByteOrder.LITTLE_ENDIAN;
+                case "ppc", "ppc64",
+                        "s390", "s390x",
+                        "sparc", "sparcv9" -> ByteOrder.BIG_ENDIAN;
+                default -> null;
+            };
+        }
```

Comment:
> To add to Jim's comment, the other thing we think about here is mapping the value of the ModuleTarget attribute to endianness, meaning don't parse it to extract the architecture.

So that would then mean updating the build file where we create these jmod file(s) to use a value for `--target-platform` that even includes the endianness in the string value, example: `aix-ppc64-big` and `macos-aarch64-little`. So a change here https://github.com/openjdk/jdk/blob/master/make/CreateJmods.gmk#L252. Did I understand it correctly?